eng/tools/generate-doc/index.js

```diff
@@ -0,0 +1,333 @@
+                      "/" +
+                      checks.version +
+                      " ./src";
+                  }
+
+                  let typedocProcess;
+                  let commandRun = [];
+                  commandRun.push("typedoc");
+                  commandRun.push({
+                    cwd: eachPackagePath,
+                    shell: true
+                  });
+                  if (checks.typedocPresent) {
+                    commandRun.push([
+                      docOutputFolder,
+                      '--theme "../../../eng/tools/generate-doc/theme/default"',
+                      "--ignoreCompilerErrors"
+                    ]);
+                  } else {
+                    commandRun.push([
+                      '--theme "../../../eng/tools/generate-doc/theme/default"',
+                      "--excludePrivate",
+                      "--excludeNotExported",
+                      '--exclude "node_modules/**/*"',
+                      "--ignoreCompilerErrors",
+                      "--mode file",
+                      docOutputFolder
+                    ]);
+                  }
+                  commandList.push(commandRun);
+                  if (generateIndexWithTemplate) {
+                    /* Adding package to packageList for the template index generation */
+                    indexPackageList.push(eachPackage);
+                  }
+                } else {
+                  console.log(
+                    "...SKIPPING Since src folder could not be found....."
+                  );
+                }
+              } else {
+                //console.log("...SKIPPING Since package is either not sdkType client");
+              }
+            } else {
+              console.log("...SKIPPING Since package marked as private...");
+            }
+          }
+        } //end-for each-package
+
+        /* Adding service entry for the template index generation */
+        serviceList.push({ name: eachService, packageList: indexPackageList });
+      }
+    } else {
+      //console.log("...SKIPPING Since service doesn't satisfy one of the 3 condition checks...");
+    }
+  } // end-for ServiceFolders
+
+
+  var plimitPromises = [];
+  const limit = pLimit(20);
+  for (const commandRun of commandList) {
+    const promise = limit(()=> new Promise(async (res, rej) => {
+
+      let typedocProcess = childProcess.spawn(commandRun[0], commandRun[2], commandRun[1]);
+      let stdOut = "";
+      let stdErr = "";
+      typedocProcess.on("close", code => {
+        res({ code, stdOut, stdErr })
+      });
+
+      typedocProcess.stdout.on("data", data => (stdOut = stdOut + data.toString()));
+      typedocProcess.stderr.on("data", err => (stdErr = stdErr));
+
+    }));
+    plimitPromises.push(promise);
+  }
+  try {
+    const results = await Promise.all(plimitPromises);
+    for (let item of results) {
+      console.log(item.stdOut);
+      if (item.code !== 0) console.log(item.stdErr);
```

Comment:
If a process exits with an error code (e.g. not 0) today this code will log the `stdErr` out to the console in the middle of other outputs. It won't highlight or place the error near the bottom of the output where a user can easily find it. This function also appears to swallow exceptions. The pipeline could succeed even if one or more typedocs processes failed because this function does not throw an error when a process it launches fails. 

An alternative approach could look like: 
```js
const results = await Promise.all(plimitPromises);

for (let item of results) { 
  console.log(item.stdOut);
  if (item.stdErr) {
    console.error(item.stdErr);
  }
if (item.code !== 0) { 
  console.error("process failed");  // maybe return the command in the promise
  process.exitCode = 1;
}
...
}
```

process.exitCode docs: https://nodejs.org/api/process.html#process_process_exitcode

Reply:
done